test/model/Report/ReportComparatorTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoSystemStatus\test\model\Report;
+
+use common_report_Report as Report;
+use oat\generis\test\TestCase;
+use oat\taoSystemStatus\model\Check\CheckInterface;
+use oat\taoSystemStatus\model\Report\ReportComparator;
+
+/**
+ * Class ReportComparatorTest
+ * @package oat\taoSystemStatus\test\model\SystemStatus
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ */
+class ReportComparatorTest extends TestCase
+{
+
+    public function testGetDegradations()
+    {
+        $oldReport = $this->getInitialReport();
+        $newReport = $this->getNewReport();
+
+        $expected = new Report(Report::TYPE_WARNING, 'Service degradations found');
+        $expected->add($this->getReport(Report::TYPE_WARNING, 'Check A warning', 'A'));
+        $expected->add($this->getReport(Report::TYPE_ERROR, 'Check E error', 'E'));
+        $expected->add($this->getReport(Report::TYPE_ERROR, 'Check G error', 'G'));
+
+        $comparator = new ReportComparator($oldReport, $newReport);
+        $this->assertEquals($expected, $comparator->getDegradations());
+
+        //No degradations - empty report
+        $comparator = new ReportComparator($oldReport, $oldReport);
+        $this->assertFalse($comparator->getDegradations()->hasChildren());
+    }
+
+    public function testGetRestorations()
+    {
+        $oldReport = $this->getInitialReport();
+        $newReport = $this->getNewReport();
+
+        $expected = new Report(Report::TYPE_SUCCESS, 'Service restorations found');
```

Comment:
Can I expect a `warning` report if no restoration is found? `info` would be more obvious IMO.